test/test_nestedtensor.py

```diff
@@ -8377,16 +8377,6 @@ def f(values, offsets):
         sample_match_fn=lambda device, sample: ("noncontig_holes" in sample.name),
         name="broken_unflatten_backward",
     ),
-    # -> CPU device conversion backwards is broken
-    XFailRule(
-        error_type=RuntimeError,
-        error_msg="Unknown layout in record_stream_any_impl",
-        op_match_fn=lambda device, op: (op.full_name == "to"),
-        sample_match_fn=lambda device, sample: (
-            sample.kwargs.get("device", None) == "cpu"
-        ),
-        name="broken_to_backward",
-    ),
```

Comment:
We no longer fail because we no longer call record_stream for case 3/ .to case. 
We want to call var.record_stream(accum_stream) when the accum stream is different from the "sync stream"
The "sync stream" prior to this PR was the default stream on the var device. With this PR, we consider the "sync stream" to be the producer stream.